frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -232,6 +233,16 @@ export const getRouteData = (route: K8sResourceKind, resource: K8sResourceKind)
   return routeData;
 };
 
+const getBuildOption = (buildConfig: K8sResourceKind, pipeline: PipelineKind) => {
+  if (buildConfig) {
+    return BuildOptions.BUILDS;
+  }
+  if (pipeline) {
+    return BuildOptions.PIPELINES;
+  }
```

Comment:
Since we show/prefer Pipelines over Builds on same pages, I would expect that we use a different order here, or?

Reply:
To understand the condition where we must default to Pipelines, we require some values of the Formik props to be set. Only then, can we make a decision on whether to go with the Pipelines or the other options? 
Currently, in this file, we are receiving the default values of Formik, therefore cannot make the decision here.
Therefore I have called the `useEffect` where we already have the values set to make the decision.